### DISCUSSION TITLE
Feature request: possibility to create own assets (rides, music, ...) for use in OpenRCT2

### DISCUSSION THREAD

**Discussion by ghost** [OpenRCT2/OpenRCT2#23081] [Feature requests] [answered]
Hi

I would like to request the possibility to create own assets (rides, music, ...) which can be used in the game.
So for this is needed:

- a format how to store them
- what existing tools to use or tools provided by the OpenRCT2 team?
- preferred a completely open source pipeline including free open source tools
- documentation that explains by giving an example
- maybe a website where the assets can be shared (similar too https://openrct2plugins.org/)

**Comment by karst** [comment, marked as answer]
Hia,
good news! This already exists :)
The parkobj is an everexpanding format for this of which you can already create your own custom
- Rides vehicles
- Ride Music
- Path surface
- Path supports/railings
- Scenery
- Themes/Scenery Groups
- Palettes

And in the pipeline are:
- Track types
- Ride type descriptors
- Peeps/Entertainers
- Campaigns/Scenario packs

Sadly we do not have a website yet to share these on, so they mostly float around on discord and New Element. 
If you download a park that has these files they will automatically be grabbed from those parks, though.

You can create your own assets using Oli's blender tool available on our discord and you can generate custom music assets using: https://goes.rctspace.com/music

**Comment by ghost** [comment]
Will the tools be released soon or are it in-game tools of OpenRCT2 and will they appear in an update?
I'm very interested.

**Reply by karst** [reply]
They aren't in game tools. Oli's tool is for Blender, and Gymnasiast's tool is on the internet.

**Comment by X123M3-256** [comment]
This was a feature of the vanilla game, and OpenRCT2 has significantly expanded the capabilities. I believe the plan is to eventually move all assets to objects but this should be considered a long term goal and I wouldn't expect it anytime soon.

The vanilla game supports the following custom asset files:

* Scenery items
* Ride vehicles
* Paths
* Park entrances
* Water palettes

OpenRCT2 additionally supports:

* Path railings
* Terrain styles
* Station styles
* Ride music

Major things which are still hardcoded include track types and staff types. I wouldn't expect these to be made into objects soon, it will take a lot of work.

**Reply by ghost** [reply]
Can you tell me how I can make the custom assets that are already supported in original rct2?
I never knew this was possible when I played the original game.

**Reply by X123M3-256** [reply]
The most popular tool for creating scenery items for the original game was the object editor, which can be found here https://www.nedesigns.com/utilities/. There is a tutorial explaining the "old school" process for creating custom rides here https://ae.rctspace.com/tutorial/index.htm

Note that these tools have major limitations, they were all written in the years after the game came out and are no longer updated. The vanilla game had a lot of capabilities when it comes to custom assets that were never taken advantage of. There's a lot of things that you can do in vanilla but which none of the available tools support (or allow for only with great difficulty).

Nowadays any people are using newer tools which were developed for OpenRCT2. The downside is that they produce asset files which will only work with OpenRCT2 and do not work with vanilla. Tolsimir at NE has been working on a modernized version of the object editor, Oli414 has a Blender plugin that can create custom scenery and I have a tool specifically aimed at rendering coaster sprites. Gymnasiast has made a tool for palettes and another one for ride music.

The OpenRCT2 asset format is just a zip file, so unlike the original you don't actually *need* any special tools to create one (though it does help).

**Reply by ghost** [reply]
Can you maybe give all links for the tools?

Are there tutorials?

Are there plans to make a website with all the tools ,tutorials and a way to share assets?
Otherwise I don't think people will start making things.

**Comment by 733737** [comment]
The point that was made is that making custom content has been possible for over 20 years. It isn't simple, and there are plenty of examples on nedesigns. Right now, it is the easiest it has ever been to make custom water palettes and ride music, but there is a process that you have to follow.
